packages/babel-parser/src/plugins/jsx/index.js

```diff
@@ -133,10 +133,10 @@ export default (superClass: Class<Parser>): Class<Parser> =>
               const htmlEntity =
                 ch === charCodes.rightCurlyBrace ? "&rbrace;" : "&gt;";
               const char = this.input[this.state.pos];
-              this.raise(
-                this.state.pos,
-                `Unexpected token \`${char}\`. Did you mean \`${htmlEntity}\` or \`{'${char}'}\`?`,
-              );
+              this.raise(this.state.pos, {
+                code: "UnexpectedToken",
```

Comment:
Should we prefix error code with `BABEL_PARSER` like we did in `BABEL_PARSER_SOURCETYPE_MODULE_REQUIRED`? Should we use SNAKE_CASE instead?

Reply:
Or maybe we could always set `code` to `BABEL_PARSER_ERROR`, and add a new property (like `errorCode`, or `parserError`) using these new codes we are introducing.

Reply:
I prefer a prefixed error code but don't have a strong opinion. The Node.js recommends `error.code` to be the identifier of the error. Conventionally they use SNAKE_CASE. 

https://nodejs.org/api/errors.html#errors_error_code

I would like to maintain the same stability policy as Node.js: The `error.code` thrown from `@babel/parser` is considered stable: It will only be changed in a major releases. While we will keep polishing the `error.message`.

Reply:
Yeah but I wonder how useful these specific error messages are.

Except for a few (and they already have specific codes), I think often I would want do do something like this:
```js
try {
  somethingThatCallsBabel();
} catch (e) {
  if (e.code === "BABEL_PARSER_ERROR") {
    console.warn("The file contains invalid syntax!");
  } else if (e.code === "BABEL_PARSER_MISSING_PLUGIN") {
    console.warn("You might want to enable a parser plugin!");
  } else if (e.code === "BABEL_PARSER_SOURCETYPE_MODULE_REQUIRED") {
    console.warn("You are parsing a module as a script!");
  } else if (e.code === "ENOENT") {
    console.warn("You are trying to parse a non-existing file!");
  } else {
    throw e;
  }
}
```

Imho the specific syntax error is a detail at a deeper leverl.

> I would like to maintain the same stability policy as Node.js: The `error.code` thrown from `@babel/parser` is considered stable

+1

Reply:
The new example looks good to me. In that case we may consider place current detailed error code to `syntaxError` or `reasonCode`.

I think `BABEL_PARSER_ERROR` is too general, maybe `BABEL_PARSER_INVALID_SYNTAX` or `BABEL_PARSER_SYNTAX_ERROR`?

Reply:
`BABEL_PARSER_SYNTAX_ERROR` looks good.

Reply:
> I would like to maintain the same stability policy as Node.js: The error.code thrown from @babel/parser is considered stable

:+1:

To summarize, the interface will look like this:

```js
```

Reply:
Yes :+1: